omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentController1_10Test.java

```diff
@@ -247,6 +248,34 @@ public void postAttachment_shouldUploadFile() throws Exception {
 			Assert.assertArrayEquals(randomData, (byte[]) complexData.getData());
 			Assert.assertNull(obs.getEncounter());
 		}
+		// File upload with Encounter
+		{
+			// Setup
+			String fileName = "testFile3.dat";
+			Patient patient = Context.getPatientService().getPatient(2);
+			Encounter encounter = testHelper.getTestEncounter();
+			
+			MockMultipartHttpServletRequest request = newUploadRequest(getURI());
+			MockMultipartFile file = new MockMultipartFile("file", fileName, "application/octet-stream", randomData);
+			
+			request.addFile(file);
+			request.addParameter("patient", patient.getUuid());
+			request.addParameter("encounter", encounter.getUuid());
+			request.addParameter("fileCaption", fileCaption);
+			
+			// Replay
+			SimpleObject response = deserialize(handle(request));
+			
+			Obs obs = Context.getObsService().getObsByUuid((String) response.get("uuid"));
+			Obs complexObs = Context.getObsService().getComplexObs(obs.getObsId(), null);
+			ComplexData complexData = complexObs.getComplexData();
+			
+			// Verify
+			Assert.assertEquals(obs.getComment(), fileCaption);
+			Assert.assertEquals(complexData.getTitle(), fileName);
+			Assert.assertArrayEquals(randomData, (byte[]) complexData.getData());
+			Assert.assertNotNull(obs.getEncounter());
```

Comment:
You should verify that it is the expected encounter, so the one whose UUID is `encounter.getUuid()`.